parsemsgpack.m

```diff
@@ -3,13 +3,14 @@
 %    reads BYTES as msgpack data, and creates Matlab data structures
 %    from it. The number of bytes consumed by the parsemsgpack call 
 %    is returned in the variable IDX.
-%    - strings are converted to strings
-%    - numbers are converted to appropriate numeric values
-%    - true, false are converted to logical 1, 0
-%    - nil is converted to []
-%    - arrays are converted to cell arrays, but converting to "normal" arrays is also possible. Look at "parsearray" and change commented lines if it suits your specific serialization.
-%    - maps are converted to structs, because in most of the cases this will be better that using containers.Map. Look at "parsemap" and change commented lines by convenience.
-%    - in case you have a specific mapping, you can add it to the function "replaceMsgPackKey".
+%    - Strings are converted to strings.
+%    - Numbers are converted to appropriate numeric values.
+%    - True, false are converted to logical 1, 0.
+%    - Nil is converted to [].
+%    - Arrays are converted to cell arrays, but converting to "normal" arrays is also possible and probably faster.
```

Comment:
This is probably false in most cases! Arrays can ONLY be used for homogenous data. `[1, "this", {"a":"b"}]` can not be represented as a Matlab array. We can not assume that a MsgPack Array only contains values of one type. Or at least, we'd need additional machinery to first check the type of each array entry, and then select an array of the appropriate sum type if there is one. Doing this would defeat the purpose of performance optimizations, however.